mango/spotmarketoperations.py

```diff
@@ -55,7 +55,7 @@ def __init__(self, context: Context, wallet: Wallet, spot_market: SpotMarket,
         self.market_index: int = market_index
         self.fee_discount_token_address: PublicKey = fee_discount_token_address
 
-        self.open_orders_address: typing.Optional[PublicKey] = self.account.spot_open_orders_by_index[self.market_index]
+        self.open_orders_address: typing.Optional[PublicKey] = self.spot_market.derive_open_orders_address(self.context, self.account)
```

Comment:
While the derived open orders address is preferred, older accounts could have a different address in use for the open orders account. Probably only fall back on derived address when there is no address yet?

(I haven't read this deeply, @OpinionatedGeek would know)